07-data_pipelines.Rmd

```diff
@@ -80,6 +80,75 @@ The repo should include:
     e.g. See: https://github.com/moj-analytical-services/airflow-enforcement-data-engineering/blob/master/iam_policy.json
     for listing, reading and writing to S3 buckets.
 
+
+## Test pipeline in your own Airflow sandbox
+
+You can deploy your own sandboxed version of Airflow to test DAGs and docker images on.
+
+### Start Airflow sandbox
+
+Click the `Deploy` button next to `airflow-sqlite` in the Analytical tools section of the control panel. This should take a few minutes to deploy. Then click the `Open` button to open your Airflow sandbox in a new window.
+
+This will create an `airflow` directory in your home directory on the Analytical Platform. In this airflow directory there are three more directories `db`, `dags` and `logs`. The `dags` directory is where you will store your test DAG files.
+
+### Create a test DAG
+
+Using JupyterLab on the Analytical Platform create a python file in the `airflow/dags` directory in your home directory in the Analytical Platform (eg. `test_dag.py`). Airflow will automatically scan this directory for DAG files every three minutes.
+
+See above for creating a Docker image to run the DAG if you are using `KubernetesPodOperator`. Concourse will automatically build images of PR's to GitHub Airflow repos and will be tagged with the name of the PR.
+
+Example DAG file contents:
+
+
+    from datetime import datetime
+
+    from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+    from airflow.utils.log.logging_mixin import LoggingMixin
+    from airflow.models import DAG
+
+    log = LoggingMixin().log
+
+    args = {"owner": "airflow", "start_date": datetime(2019, 3, 18)}
+
+    dag = DAG(
+        dag_id="example_kubernetes_operator_assume_role",
+        default_args=args,
+        schedule_interval=None,
+    )
+
+    k = KubernetesPodOperator(
+        namespace="user-###YOUR_GITHUB_USERNAME###",
+        image="governmentpaas/awscli:latest",
+        cmds=["aws", "--debug"],
+        arguments=["s3api", "list-objects", "--bucket", "###YOUR_TEST_BUCKET###"],
+        env_vars={
+            "AWS_METADATA_SERVICE_TIMEOUT": "60",
+            "AWS_METADATA_SERVICE_NUM_ATTEMPTS": "5",
+        },
+        labels={"foo": "bar"},
+        name="airflow-test-pod",
+        in_cluster=True,
+        task_id="task",
+        get_logs=True,
+        dag=dag,
+        is_delete_operator_pod=True,
+        annotations={"iam.amazonaws.com/role": "alpha_user_###YOUR_GITHUB_USERNAME###"},
+    )
+
+
+
+You will have to set the DAG namespace to your namespace and the AWS role to your role, see `###YOUR_GITHUB_USERNAME###` in the above example code.
+
+You will also need to update the image with your DAG image from AWS ECR and add any environment variables etc. as you would with a normal DAG.
+
+### IMPORTANT
```

Comment:
Could you change this to something like `Important considerations`?

Reply:
Yeah sure. Do you think in all caps? It is very important that people read and understand them.